ml-agents/mlagents/envs/rpc_communicator.py

```diff
@@ -41,20 +40,32 @@ def __init__(self, worker_id=0,
         self.server = None
         self.unity_to_external = None
         self.is_open = False
+        self.create_server()
+
+    def create_server(self):
+        # Attempt to open a socket, failing if the port is in use.
+        self.check_port(self.port)
 
-    def initialize(self, inputs: UnityInput) -> UnityOutput:
         try:
             # Establish communication grpc
             self.server = grpc.server(ThreadPoolExecutor(max_workers=10))
             self.unity_to_external = UnityToExternalServicerImplementation()
             add_UnityToExternalServicer_to_server(self.unity_to_external, self.server)
-            self.server.add_insecure_port('[::]:'+str(self.port))
+            self.server.add_insecure_port('[::]:' + str(self.port))
             self.server.start()
-        except :
-            raise UnityTimeOutException(
-                "Couldn't start socket communication because worker number {} is still in use. "
-                "You may need to manually close a previously opened environment "
-                "or use a different worker number.".format(str(self.worker_id)))
+        except:
+            raise UnityWorkerInUseException(self.worker_id)
+
+    def check_port(self, port):
```

Comment:
Can you add a little comment for this method as well ?

Reply:
Done.